src/app/content/utils/canonicalUrl.ts

```diff
@@ -9,22 +9,31 @@ export async function getCanonicalUrlParams(
   archiveLoader: AppServices['archiveLoader'],
   osWebLoader: AppServices['osWebLoader'],
   bookId: string,
-  pageShortId: string
+  pageShortId: string,
+  bookVersion?: string
 ) {
   const getBook = makeUnifiedBookLoader(archiveLoader, osWebLoader);
-  const canonicals = [
-    ...(CANONICAL_MAP[bookId] || []),
-    bookId, // use the current book as a last resort
-  ].filter((id) => !!BOOKS[id]);
 
-  for (const id of canonicals) {
-    const version = BOOKS[id].defaultVersion;
+  const canonicals = [...(CANONICAL_MAP[bookId] || []), bookId] // use the current book as a last resort
+    .filter((id) => !!BOOKS[id])
+    .map((id) => ({
+      id,
+      version: bookVersion && id === bookId
+        ? bookVersion
+        : BOOKS[id].defaultVersion,
+    }));
+
+  for (const {id, version} of canonicals) {
 
     const canonicalBook = await getBook(id, version);
     const treeSection = findArchiveTreeNode(canonicalBook.tree, pageShortId);
     if (treeSection) {
       const pageInBook = assertDefined(treeSection.slug, 'Expected page to have slug.');
-      return {book: canonicalBook.slug, page: pageInBook};
+      const params = {book: canonicalBook.slug, page: pageInBook};
+
+      return version === BOOKS[id].defaultVersion || id !== bookId
```

Comment:
isnt' there a utility that does this?

Reply:
`getBookPageUrlAndParams` does but it will throw if tree doesn't have the section which i don't think is desired behavior here.